### PR TITLE
Digital twin

### PR DESCRIPTION
**Pull request by nolanporter77**

### CODE REVIEW

File: config.yaml
Comment (adv3429) [inline config.yaml:1]: Do we want a config option for center frame, left frame, right frame, etc.?

Comment (adv3429) [reply config.yaml:1]: ^ this could be a feature for the future after P1

**Review by adv3429 — APPROVED** [review]
Good for P1 director. nice work :)

File: digital_twin/digital_twin.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        if delta_azimuth != 0:
+            self.rotate_azimuth(delta_azimuth)
+        if delta_altitude != 0:
+            self.rotate_altitude(delta_altitude)
+
+    def rotate_azimuth(self, deg):
+        """
+        Method to rotate the base of the digital twin
+        """
+        joint_index = 0  # Azimuth joint index
```
Comment (johnaflory) [inline digital_twin/digital_twin.py:43]: Lets define these as global const variables

File: directors/basic_director.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        #print("Start of test")
+
+
+        if len(frame) > 0:
+            acceptable_box_left, acceptable_box_top, acceptable_box_right, acceptable_box_bottom = self.calculate_acceptable_box();
+
+            #Calculate where the middle point of the bounding box lies in relation to the box
+            # Unpack bounding box
+            #Right now I am going to assume we only want the first face
+            first_face = frame[0]
```
Comment (johnaflory) [inline directors/basic_director.py:62]: I'm not sure if faces are ordered by confidence it is a face. Future work should be done to make sure we only track the speakers face.

File: digital_twin/digital_twin.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
+import pybullet as p
```
Comment (DevanTurtle7) [inline digital_twin/digital_twin.py:1]: Nitpick but I think I'd like a more descriptive name

File: requirements.txt [hunk 4]
```diff
@@ -4,4 +4,6 @@ websocket-client==1.8.0
 abcmeta==2.2.0
```
Comment (DevanTurtle7) [inline requirements.txt:4]: Thanks for updating

File: config.yaml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+acceptable_box_percent: 0.5
```
Comment (DevanTurtle7) [inline config.yaml:1]: Glad we used yaml for config

**Review by DevanTurtle7 — APPROVED** [review]
Looks good, small nitpick

File: directors/basic_director.py (hunk 2 above)
Comment (adv3429) [reply directors/basic_director.py]: Added a TODO to come back to this. This is an assumption we'll make for now for the demo

**Comment by DevanTurtle7** [comment]
closes #28 
closes #37 
closes #36

**Comment by nolanporter77** [comment]
PR Confidence: 2
Director implemented, but no way to test with actual robot

**Pull request merged**